File: flybanner/src/main/java/com/xia/flybanner/listener/OnPageChangeListener.java
package com.xia.flybanner.listener;

import androidx.recyclerview.widget.RecyclerView;

/**
 * @author weixia
 * @date 2019/4/16.
 */
public interface OnPageChangeListener {

    void onScrollStateChanged(RecyclerView recyclerView, int newState);

    void onScrolled(RecyclerView recyclerView, int dx, int dy);

    void onPageSelected(int index, boolean isLastPage);
}

File: flybanner/src/main/java/com/xia/flybanner/listener/OnPageChangeListener.kt
package com.xia.flybanner.listener

import androidx.recyclerview.widget.RecyclerView

/**
 * @author weixia
 * @date 2019/4/16.
 */
interface OnPageChangeListener {

    fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int)

    fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int)

    fun onPageSelected(index: Int, isLastPage: Boolean)
}
